core/src/main/java/org/apache/iceberg/MetadataUpdate.java

```diff
@@ -38,6 +39,23 @@ default void applyTo(ViewMetadata.Builder viewMetadataBuilder) {
         String.format("Cannot apply update %s to a view", this.getClass().getSimpleName()));
   }
 
+  class SetKekCache implements MetadataUpdate {
```

Comment:
TODO: find better way of propagating KEK cache than having it be a `MetadataUpdate` in a potential `UpdateTableRequest`.